src/ast/mod.rs

```diff
@@ -3027,6 +3027,8 @@ pub enum Statement {
         statement: Box<Statement>,
         /// Optional output format of explain
         format: Option<AnalyzeFormat>,
+        /// Postgres style utility options, `(analyze, verbose true)`
+        options: Option<UtilityOptionList>,
```

Comment:
UtilityOptionList is undocumented but appears in PostgreSQL parser rules and is used in statements like EXPLAIN, VACUUM, and CLUSTER. It consistently takes the form (analyze true, verbose).We might be able to reuse it in other statement implementations. (I think we don't need to implement Display ( xxx ) in this way)

Reply:
hmm will the representation always be a `Vec<UtilityOption>` in all usages? i.e. do we expect any new fields to be added to this struct?
```rust
pub struct UtilityOptionList {
    pub options: Vec<UtilityOption>,
}
```
If we don't, then the struct doesn't seem to provide much over using a plain Vec was what I meant - its still reusable in the same way. And with a vec we dont need to implement display at all (I don't think?) since we have `display_comma_separated`

Reply:
By keeping UtilityOptionList, we can avoid having each statement implementation handle the Display trait for printing options in the format `(optname optarg, optname, ...)`.

> hmm will the representation always be a `Vec<UtilityOption>` in all usages? i.e. do we expect any new fields to be added to this struct?
> 
> ```rust
> pub struct UtilityOptionList {
>     pub options: Vec<UtilityOption>,
> }
> ```
> 
> If we don't, then the struct doesn't seem to provide much over using a plain Vec was what I meant - its still reusable in the same way. And with a vec we dont need to implement display at all (I don't think?) since we have `display_comma_separated`

Reply:
> By keeping UtilityOptionList, we can avoid having each statement implementation handle the Display trait for printing options in the format (optname optarg, optname, ...)

We might be misunderstanding each other.

To clarify the display behavior I mean is that all that changes between both approaches is that each Statement need to call:

with `UtilityOptionList`

```rust
write!(f, "{utility_option_list}");
```

vs
with `Vec<UtilityOption>`

```rust
write!(f, "({})", display_comma_separated(options));
```

And everything else is exactly the same for statements - unless I'm missing something.

Can you provide a code example of what you mean, i.e. how would the display trait differ and what's the pro/con?

Reply:
Sorry, it should be the same. I missed the display_comma_separated function.

Removed the UtilityOptionList and merge two doc and move to the `UtilityOption`.
Thank you